.github/workflows/promote-to-release-channel-with-config.yaml

```diff
@@ -0,0 +1,149 @@
+name: "Promote to release channel with config"
+
+on:
+  workflow_dispatch:
+    inputs:
+      new_warden_tag:
+        description: 'New warden tag (`x.x.x`)'
+        default: ""
+        required: true
+      warden_skr_config_version:
+        description: 'Warden SKR config version (`y.y`)'
+        default: ""
+        required: true
+      channel:
+        description: 'Module channel'
+        default: "fast"
+        required: true
+
+env:
+  MODULE_VERSION: ${{ github.event.inputs.new_warden_tag }}
+  CHANNEL: ${{ github.event.inputs.channel }}
+  # needed by gh cli for GitHub enterprise
+  GH_ENTERPRISE_TOKEN: ${{ secrets.GH_BOT_TOKEN }}
+  WARDEN_SKR_OVERRIDES_REPO_URL: ${{ secrets.WARDEN_SKR_OVERRIDES_REPO_URL }}
+  BOT_USERNAME: kyma-otter-serviceuser
+  BOT_EMAIL: dl_545734d8fd84a02c39000002@global.corp.sap
+
+jobs:
+
+  upgrade-images:
+    name: Bump warden images references
+    needs: verify-head-status
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      
+      - name: Bump sec-scanners-config.yaml
+        run: ./.github/scripts/upgrade-sec-scanners-config.sh
+        env:
+          IMG_VERSION: ${{ github.event.inputs.new_warden_tag }}
+      
+      - name: Bump values.yaml
+        run: |
+          make replace-chart-images
+        env:
+          IMG_DIRECTORY: "prod"
+          IMG_VERSION: ${{ github.event.inputs.new_warden_tag }}
+      
+      - name: Bump chart version
+        run: ./.github/scripts/upgrade-chart-version.sh
+        env:
+          CHART_VERSION: ${{ github.event.inputs.new_warden_tag }}
+      
+      - name: Commit&Push
+        run: |
+          git config --local user.email "otter-releaser@otter-releaser.noreply.github.com"
+          git config --local user.name "otter-releaser"
+
+          git add .
+          git commit --allow-empty -m "upgrade dependencies"
+          git push origin ${{ github.ref_name }}
+
+  create-tag:
+    name: Create tag
+    needs: upgrade-images
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.ref_name }}
+
+      - name: Create lightweight tag
+        run: |
+          git tag ${{ github.event.inputs.new_warden_tag }}
+          git push origin ${{ github.event.inputs.new_warden_tag }}
+
+      - name: Verify prow release jobs
+        run: ./.github/scripts/verify-release-builds.sh ${{ github.ref_name }}
+
+  promote-warden:
+    name: Render warden with skr config
+    needs: create-tag
+    runs-on: ubuntu-latest
+  
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
The `actions/checkout@v4` action by default checkouts to the SHA in the action context that triggered the workflow. In this case, we need to checkout to new SHA created by push in the first job (`upgrade-images`). I would suggest to add `ref: ${{ github.ref_name }}` arg.